vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/ConsumableResourceChecker.java

```diff
@@ -100,9 +100,10 @@ public Optional<String> unavailable() {
                   });
       if (error.isPresent()) {
         updateBlockedResource(error.get());
-        // Skip the last one, because it already failed, so we don't have to release any resources.
-        while (--i >= 0) {
-          resourceBrokers.get(i).second().release(workflow, workflowVersion, vidarrId);
+        // Each resource performs a check and only releases resources if they've been acquired
+        for (int ix=0; ix < resourceBrokers.size(); ix++){
```

Comment:
I don't understand this change. By draining `i` in the old method, we only run `release()` on resources that previously returned AVAILABLE (which is effectively the condition for `i` counting up). Now we start again from the beginning and `release()` resources that `i` never got to count up to, why? It seems like wasted computational effort

Reply:
Because `release()` is how the resource is being adding to the waitinglist and counting up to `i` ensures it's called every time.
This changes makes it so that no matter which resource throttles the run, it's added to waiting and will be relaunched in the correct order of priority. There's no error if a resource isn't being held when `release()` is called, but you're right the iteration is more computational effort then before.